app/uk/gov/hmrc/gform/models/VisibilityFilter.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.models
+
+import uk.gov.hmrc.gform.models.optics.{ DataOrigin, FormModelVisibilityOptics }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ FormComponent, FormPhase, Group, IsGroup, IsRevealingChoice, RevealingChoice }
+
+class VisibilityFilter[D <: DataOrigin, P <: PageMode](
```

Comment:
FormComponentVisibilityFilter?

Reply:
What about 
```
app/uk/gov/hmrc/gform/models/formcomponent/VisibilityFilter.scala
```
instead?

Reply:
Nevermind, `FormComponentVisibilityFilter` is it.